editor/editor_dock_manager.cpp

```diff
@@ -180,7 +180,17 @@ void EditorDockManager::_update_docks_menu() {
 }
 
 void EditorDockManager::_docks_menu_option(int p_id) {
-	focus_dock(docks_menu_docks[p_id]);
+	Control *dock = docks_menu_docks[p_id];
+	ERR_FAIL_NULL(dock);
+	ERR_FAIL_COND_MSG(!all_docks.has(dock), vformat("Menu option for unknown dock '%s'.", dock->get_name()));
+	if (all_docks[dock].enabled && all_docks[dock].open) {
+		docks_menu->hide();
```

Comment:
Hiding parent menu is enough. I think you can assume it exists and do ERR_FAIL_NULL if it doesn't for some reason.